Classes/WPEditorViewController.m

```diff
@@ -1555,13 +1589,25 @@ - (void)editorView:(WPEditorView*)editorView
 - (void)editorView:(WPEditorView*)editorView
       fieldFocused:(WPEditorField*)field
 {
-    if (!field || field == self.editorView.titleField) {
+    if (field == self.editorView.titleField) {
         [self.toolbarView enableToolbarItems:NO shouldShowSourceButton:YES];
     } else if (field == self.editorView.contentField) {
         [self.toolbarView enableToolbarItems:YES shouldShowSourceButton:YES];
     }
 }
 
+- (void)editorView:(WPEditorView*)editorView sourceFieldFocused:(UIView*)view
+{
+    [self.toolbarView enableToolbarItems:NO shouldShowSourceButton:YES];
+    
+    // Enable the toolbar if the HTML editor has focus
+    if ([view isKindOfClass:[UITextView class]]) {
```

Comment:
This works for me. I think my obtuse approach was the result of late-night coding. :wink: